client/src/Core/Components/Navbar/NavbarTop.tsx

```diff
@@ -32,15 +32,21 @@ const NavbarTopInner = ({ styles }: IStylesOnly<NavbarTopStyles>): JSX.Element |
   const history = useHistory();
   const assignmentStore = useStore('assignmentStore');
   const location = useLocation();
+  const { filterStore } = useStore('microsoftLearnStore');
   
   // The following map can be extended to include other searchParams as well in future in case need be
   let uriSearchParamsMap = new Map<string, string>();
-  uriSearchParamsMap.set(pagesDisplayNames.MSLEARN, ''); // replace empty string with store state
+  uriSearchParamsMap.set(pagesDisplayNames.MSLEARN, filterStore.learnFilterUriParam); 
```

Comment:
Please see if the below is a bit better in readability?
```suggestion
  const queryParamsMap= { pagesDisplayNames.MSLEARN:  filterStore.learnFilterUriParam };
  
 function pushToHistory(item: PivotItem) {
   if(item.props && item.props.itemKey && item.props.headerText) {
     const { itemKey, headerText } = item.props;
     const queryParam = uriQueryParamsMap[headerText];
     history.push(`${itemKey}?${queryParam}`)
   }
 }
 
 function handleLinkClick(item?: PivotItem, event?:MouseEvent): void {
   event?.preventDefault();
   if(item) pushToHistory(item);
 }
```

Reply:
I have incorporated most of the above-suggested changes, but for one thing- the map declaration in `line 39`: I tried it before also, it doesn't seem to work as it is, throws off a compiler error on my end. Could be the difference between es6 and es5?

Reply:
The problem is what I'm trying to create is not a `Map` but a Javascript Object. Trying to coerce it with a map would essentially tend to fail.